src/Utilities/useGetEnvironment.ts

```diff
@@ -26,3 +28,24 @@ export const useFlagWithEphemDefault = (
   const { getEnvironment } = useChrome();
   return (getEnvironment() === 'qa' && ephemDefault) || getFlag;
 };
+
+export const useGetFeatureFlag = (flagName: string) => {
+  // If it's the Cockpit app (on-premise), always return false
+  if (isOnPremise) {
+    return false;
+  }
+  switch (flagName) {
+    case 'edgeParity.image-list':
+    case 'image-builder.snapshots.enabled':
+    case 'image-builder.firstboot.enabled':
+    case 'image-builder.wsl.enabled':
+    case 'image-builder.compliance.enabled':
+      // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
React hooks can't be used inside of conditionals, that's what was probably causing the problem with react-hooks/rules-of-hooks... I don't think we should violate this rule. Let's try to find a pattern that doesn't require disabling this rule.